Lib/test/test_asyncio/test_sendfile.py

```diff
@@ -484,8 +485,17 @@ def sendfile_native(transp, file, offset, count):
 
         srv_proto, cli_proto = self.prepare_sendfile(close_after=1024)
         with self.assertRaises(ConnectionError):
-            self.run_loop(
-                self.loop.sendfile(cli_proto.transport, self.file))
+            try:
+                self.run_loop(
+                    self.loop.sendfile(cli_proto.transport, self.file))
+            except OSError as e:
+                # macOS may raise OSError of EPROTOTYPE when writing to a
+                # socket that is in the process of closing down.
+                if e.errno == errno.EPROTOTYPE and sys.platform == "darwin":
+                    raise ConnectionError from e
```

Comment:
```suggestion
                    raise ConnectionError
```

Raising exceptions inside `except` automatically has `from e` part. See:

```python
>>> try:
...    1 / 0
... except ZeroDivisionError:
...    raise TypeError('example')
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
TypeError: example
```